src/dmn/trn_rpc_protocol_handlers_1.c

```diff
@@ -243,9 +243,14 @@ int *update_ep_1_svc(rpc_trn_endpoint_t *ep, struct svc_req *rqstp)
 		goto error;
 	}
 
-	memcpy(epval.remote_ips, ep->remote_ips.remote_ips_val,
-	       epval.nremote_ips * sizeof(epval.remote_ips[0]));
-
+	if (epval.nremote_ips > 0) {
+		memcpy(epval.remote_ips, ep->remote_ips.remote_ips_val,
+		       epval.nremote_ips * sizeof(epval.remote_ips[0]));
+	}
+	if (epval.nremote_ports > 0) {
+		memcpy(epval.remote_ports, ep->remote_ports.remote_ports_val,
+		       epval.nremote_ports * sizeof(epval.remote_ports[0]));
+	}
```

Comment:
Shouldn't the nremote_ports always equal nremote_ips? Are you making a default value. If not, I believe these two should can be combined.

Reply:
Not all the time. We can have a single simple endpoint with no remote ports. Another scenario we need to support for Kubernetes is multi-port services where the number of backends can differ from the ports.
https://kubernetes.io/docs/concepts/services-networking/service/#multi-port-services